app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -35,27 +49,49 @@
     private DrawerLayout mDrawerLayout;
     private Toolbar mToolbar;
     private NavigationView navigationView;
+    private ProgressBar mProgress;
+    private FrameLayout frameLayout;
+    private int counter;
+    private int eventsDone;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-
-
-        DataDownload download = new DataDownload();
-        download.downloadVersions();
-
+        eventsDone = 0;
         setUpToolbar();
         setUpNavDrawer();
+        frameLayout = (FrameLayout) findViewById(R.id.layout_main);
         navigationView = (NavigationView) findViewById(R.id.nav_view);
+        mProgress = (ProgressBar) findViewById(R.id.progress);
+        mProgress.setVisibility(View.VISIBLE);
+        mProgress.setIndeterminate(true);
+        Bus bus = OpenEventApp.getEventBus();
+        bus.register(this);
+        DataDownload download = new DataDownload();
```

Comment:
But it takes time right?

On Mon, Aug 10, 2015 at 3:07 AM, Manan Wason <notifications@github.com>
wrote:

> In app/src/main/java/org/fossasia/openevent/activities/MainActivity.java
> <https://github.com/fossasia/open-event-android/pull/69#discussion_r36592408>
> :
>
> >          navigationView = (NavigationView) findViewById(R.id.nav_view);
> > +        mProgress = (ProgressBar) findViewById(R.id.progress);
> > +        mProgress.setVisibility(View.VISIBLE);
> > +        mProgress.setIndeterminate(true);
> > +        Bus bus = OpenEventApp.getEventBus();
> > +        bus.register(this);
> > +        DataDownload download = new DataDownload();
>
> Basically this is done only once at startup otherwise local data is shown
> even when there is no internet.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fossasia/open-event-android/pull/69/files#r36592408>.
>



-- 
Mohit Singh KANWAL (Mr)

[image: Facebook] <http://facebook.com/mohitkanwal>  [image: Twitter]
<http://twitter.com/mohitkanwal>